.travis.yml

```diff
@@ -19,18 +19,20 @@ matrix:
   fast_finish: true
   include:
     - php: 7.1
-      env: DB=MYSQL PHPUNIT_TEST=1 PHPCS_TEST=1
+      env: DB=MYSQL PHPUNIT_TEST=1 COMPOSER_ARG=--prefer-lowest
     - php: 7.2
-      env: DB=MYSQL PHPUNIT_TEST=1
+      env: DB=MYSQL PHPUNIT_TEST=1 PHPCS_TEST=1
     - php: 7.3
       env: DB=MYSQL PDO=1 PHPUNIT_COVERAGE_TEST=1
     - php: 7.4
       env: DB=MYSQL PHPUNIT_TEST=1
+    - php: nightly
+      env: PHPUNIT_TEST=1 COMPOSER_ARG=--ignore-platform-reqs WITH_XDEBUG=1
 
 before_script:
 # Init PHP
   - phpenv rehash
-  - phpenv config-rm xdebug.ini || true
+  - if [[ ! $WITH_XDEBUG ]]; then phpenv config-rm xdebug.ini fi;
```

Comment:
We've been using phpdbg instead of xdebug for code coverage runs, is that changing? If so you'll need to update line 48

Reply:
No, it does not change. The default behaviour stays as is. This flag merely allows using XDebug if necessary. In this particular case it's used to skip xdebug removal, which is unavailable in php-nightly builds.

Reply:
`$WITH_XDEBUG` as a variable name is confusing because it's implying that php8 is actually using xdebug, when in fact it's simply "not removing it because it does not exist".  Unless you're familiar with context of this, it's very confusing.

$XDEBUG_UNAVAILABLE would be a lot clearer IMO.  You would end up with a double negative `if [[ ! $XDEBUG_UNAVAILABLE ]]`, though I think this is fine.


Also: do we even need this at all?  This doesn't not have it and it seems fine? https://github.com/silverstripe/silverstripe-admin/pull/1118/files

Reply:
`silverstripe-admin` works because it has a silent fail there (`|| true` part).
Silencing errors is an anti-pattern, so I came up with the variable `$WITH_XDEBUG` instead.
For the `nightly` build it's a separate thing and more like an excption. When PHP8 becomes stable, it'll have XDEBUG there, but I'm not sure about phpdbg though - IIRC it's been deprecated and won't make it to php8.

Reply:
ok, can we change the variable name to `$XDEBUG_UNAVAILABLE` then?  It may mean we need to update a couple of existing merged PR's that have `$WITH_XDEBUG` though I think this is still worthwhile so we don't look at this in 6 months and wonder what the heck it is ;-)

Reply:
IMHO `$WITH_XDEBUG` is more aligned with the usual software conventions as well as with PHP own flag conventions.
E.g. some of the PHP compilation flags are `--with-sqlite3` or `--with-sodium`.

`XDEBUG_UNAVAILABLE` would imply that this variable is set by something checking if it's available or not, which is not the case.

`USE_XDEBUG` also feels to me like it means something different (e.g. I would assume that tells to tests to use xdebug or something like that).

> this is still worthwhile so we don't look at this in 6 months and wonder what the heck it is ;-)

I guess if that's unclear, we would better go without any variables and revert to using `|| true`, which doesn't give any control over it, but doesn't introduce new variables.

Reply:
My confusion when I initially read this was caused by the fact that this environment variable is defined in the first place. To me it implies that we want to run xdebug for some builds. My vote would be to keep it the way it was (silent failover), but you're more actively maintaining Travis configs than I am nowadays so happy to follow your consensus.

Reply:
Reverted it back to silent. No much need in running xdebug, that's true :)